otx/mpa/modules/models/heads/custom_multi_label_non_linear_cls_head.py

```diff
@@ -41,18 +41,17 @@ def __init__(
         self.in_channels = in_channels
         self.num_classes = num_classes
         self.hid_channels = hid_channels
-        self.act = build_activation_layer(act_cfg)
         self.dropout = dropout
         self.normalized = normalized
         self.scale = scale
 
         if self.num_classes <= 0:
             raise ValueError(f"num_classes={num_classes} must be a positive integer")
 
-        self._init_layers()
+        self._init_layers(act_cfg)
 
-    def _init_layers(self):
-        modules = [nn.Linear(self.in_channels, self.hid_channels), nn.BatchNorm1d(self.hid_channels), self.act]
+    def _init_layers(self, act_cfg):
+        modules = [nn.Linear(self.in_channels, self.hid_channels), nn.BatchNorm1d(self.hid_channels), build_activation_layer(act_cfg)]
```

Comment:
Is this remedy just targetted to `multi_label_non_linear_cls_head.py`? what about `linear_cls_head`?

Reply:
In the multi-class classification we don't have such problem, since don't use PRelu activation. 
In general we need to get rid of awful implementation of state dict hooks. There are a lot of hardcoded things and it can shock  someone who reading this